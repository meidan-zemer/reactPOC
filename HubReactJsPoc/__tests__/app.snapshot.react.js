var React = require("react");
var App = require("../app/app.jsx");
var renderer = require("react-test-renderer");
var scene = {
                "sceneName": "Opening",
                "edges": [{
                    "destination": {
                        "sceneName": "World of Content",
                        "edges": [{
                            "destination": {
                                "sceneName": "Content Rec",
                                "edges": [{
                                    "destination": {
                                        "sceneName": "App Intro",
                                        "edges": [{
                                            "destination": {
                                                "sceneName": "TV Go",
                                                "edges": [{
                                                    "destination": {
                                                        "sceneName": "Wifi Hotspots",
                                                        "edges": [{
                                                            "destination": {
                                                                "sceneName": "Wifi App",
                                                                "edges": [{
                                                                    "destination": {
                                                                        "sceneName": "My Account",
                                                                        "edges": [{
                                                                                "destination": {
                                                                                    "sceneName": "CTA"
                                                                                }
                                                                            }

                                                                        ]
                                                                    }
                                                                }]
                                                            },
                                                            "conditions": [{
                                                                "variable": "No TV"
                                                            }]
                                                        }]
                                                    },
                                                    "conditions": [{
                                                        "variable": "X1 on legacy"
                                                    }, {
                                                        "variable": "Upper tier Internat"
                                                    }]
                                                }, {
                                                    "destination": {
                                                        "sceneName": "Aports App"
                                                    },
                                                    "conditions": [{
                                                        "variable": "X1 Only"
                                                    }, {
                                                        "variable": "Viewing History Sports"
                                                    }]
                                                }, {
                                                    "destination": {
                                                        "sceneName": "Kid Zone"
                                                    },
                                                    "conditions": [{
                                                        "variable": "X1 Only"
                                                    }, {
                                                        "variable": "Viewing History Kids"
                                                    }]
                                                }]

                                            },
                                            "conditions": null
                                        }]

                                    }
                                }]
                            },
                            "conditions": [{
                                "variable": "X1"
                            }]
                        }]
                    },
                    "conditions": [{
                        "variable": "TV/TV Internet"
                    }]
                }]
            };


test('Render App',function (){
   var component =renderer.create(
        <App scene={scene}/>
   )
   expect(component.toJSON()).toMatchSnapshot();
});