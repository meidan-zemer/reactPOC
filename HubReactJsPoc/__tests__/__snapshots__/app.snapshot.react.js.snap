exports[`test Render App 1`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Download JSON" />
  <ul>
    <li
      className="scene">
      <span
        className="node-management">
        <a
          className="add-child"
          href="javascript: void(0)"
          onClick={[Function]}>
          +
        </a>
      </span>
      <span
        className="scene-title">
        Opening
      </span>
      <ul>
        <li
          className="scene">
          <span
            className="node-management">
            <a
              className="add-child"
              href="javascript: void(0)"
              onClick={[Function]}>
              +
            </a>
          </span>
          <span
            className="scene-title">
            World of Content
          </span>
          <ul>
            <li
              className="scene">
              <span
                className="node-management">
                <a
                  className="add-child"
                  href="javascript: void(0)"
                  onClick={[Function]}>
                  +
                </a>
              </span>
              <span
                className="scene-title">
                Content Rec
              </span>
              <ul>
                <li
                  className="scene">
                  <span
                    className="node-management">
                    <a
                      className="add-child"
                      href="javascript: void(0)"
                      onClick={[Function]}>
                      +
                    </a>
                  </span>
                  <span
                    className="scene-title">
                    App Intro
                  </span>
                  <ul>
                    <li
                      className="scene">
                      <span
                        className="node-management">
                        <a
                          className="add-child"
                          href="javascript: void(0)"
                          onClick={[Function]}>
                          +
                        </a>
                      </span>
                      <span
                        className="scene-title">
                        TV Go
                      </span>
                      <ul>
                        <li
                          className="scene">
                          <span
                            className="node-management">
                            <a
                              className="add-child"
                              href="javascript: void(0)"
                              onClick={[Function]}>
                              +
                            </a>
                          </span>
                          <span
                            className="scene-title">
                            Wifi Hotspots
                          </span>
                          <ul>
                            <li
                              className="scene">
                              <span
                                className="node-management">
                                <a
                                  className="add-child"
                                  href="javascript: void(0)"
                                  onClick={[Function]}>
                                  +
                                </a>
                              </span>
                              <span
                                className="scene-title">
                                Wifi App
                              </span>
                              <ul>
                                <li
                                  className="scene">
                                  <span
                                    className="node-management">
                                    <a
                                      className="add-child"
                                      href="javascript: void(0)"
                                      onClick={[Function]}>
                                      +
                                    </a>
                                  </span>
                                  <span
                                    className="scene-title">
                                    My Account
                                  </span>
                                  <ul>
                                    <li
                                      className="scene">
                                      <span
                                        className="node-management">
                                        <a
                                          className="add-child"
                                          href="javascript: void(0)"
                                          onClick={[Function]}>
                                          +
                                        </a>
                                      </span>
                                      <span
                                        className="scene-title">
                                        CTA
                                      </span>
                                    </li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                        <li
                          className="scene">
                          <span
                            className="node-management">
                            <a
                              className="add-child"
                              href="javascript: void(0)"
                              onClick={[Function]}>
                              +
                            </a>
                          </span>
                          <span
                            className="scene-title">
                            Aports App
                          </span>
                        </li>
                        <li
                          className="scene">
                          <span
                            className="node-management">
                            <a
                              className="add-child"
                              href="javascript: void(0)"
                              onClick={[Function]}>
                              +
                            </a>
                          </span>
                          <span
                            className="scene-title">
                            Kid Zone
                          </span>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
